lib/knapsack_pro/extensions/rspec_extension.rb

```diff
@@ -0,0 +1,164 @@
+          configuration.filter_manager = filter_manager
+
+          configuration.knapsack__load_spec_files(test_file_paths)
+        end
+
+        # Based on:
+        # https://github.com/rspec/rspec-core/blob/f8c8880dabd8f0544a6f91d8d4c857c1bd8df903/lib/rspec/core/runner.rb#L113
+        #
+        # Option: `configuration.fail_if_no_examples`
+        #   Ignore the configuration.fail_if_no_examples option because it should be off in Queue Mode.
+        #   * Ignore the fail_if_no_examples option because in Queue Mode a late CI node can start after other CI nodes already executed tests. It is expected not to run examples in such a scenario.
+        #   * RSpec should not fail when examples are not executed for a batch of tests fetched from the Queue API. The batch could have tests that have no examples (for example, someone commented out the content of the spec file). We should fetch another batch of tests from the Queue API and keep running tests.
+        #
+        # @return [Fixnum] exit status code.
+        def knapsack__run_specs(queue_runner)
+          # Based on:
+          # https://github.com/rspec/rspec-core/blob/f8c8880dabd8f0544a6f91d8d4c857c1bd8df903/lib/rspec/core/world.rb#L53
+          ordering_strategy = configuration.ordering_registry.fetch(:global)
+          node_examples_passed = true
+
+          configuration.with_suite_hooks do
+            exit_status = configuration.reporter.report(_expected_example_count = 0) do |reporter|
+              queue_runner.with_batched_tests_from_queue do |test_file_paths|
+                knapsack__load_spec_files_batch(test_file_paths)
+
+                examples_passed = ordering_strategy.order(world.example_groups).map do |example_group|
+                  queue_runner.handle_signal!
+                  example_group.run(reporter)
+                end.all?
+
+                node_examples_passed = false unless examples_passed
+
+                knapsack__persist_example_statuses
+
+                if reporter.fail_fast_limit_met?
+                  queue_runner.log_fail_fast_limit_met
+                  break
+                end
+              end
+            end
+
+            exit_code(node_examples_passed)
+          end
+        end
+
+        # Based on:
+        # https://github.com/rspec/rspec-core/blob/f8c8880dabd8f0544a6f91d8d4c857c1bd8df903/lib/rspec/core/runner.rb#L90
+        def knapsack__persist_example_statuses
+          persist_example_statuses
+        end
+      end
+
+      module Configuration
+        # Based on:
+        # https://github.com/rspec/rspec-core/blob/f8c8880dabd8f0544a6f91d8d4c857c1bd8df903/lib/rspec/core/configuration.rb#L1619
+        def knapsack__load_spec_files(test_file_paths)
+          batch_of_files_to_run = get_files_to_run(test_file_paths)
+          batch_of_files_to_run.each do |f|
+            file = File.expand_path(f)
+            load_file_handling_errors(:load, file)
+            loaded_spec_files << file
+          end
```

Comment:
It has impact only on https://github.com/rspec/rspec-core/blob/5c8d1799f54d231bf97d9a0e558ffa3938a05932/lib/rspec/core/configuration.rb#L2329 if it's true. But we don't care about that.

Integration tests are still green after I added `@spec_files_loaded = true`.

The only argument not to add `@spec_files_loaded = true` is that we call `knapsack__load_spec_files` multiple times (for each batch of tests). So it's not entirely true that `@spec_files_loaded` is `true` because there will be more tests loaded in the next batch so I did not added the `@spec_files_loaded = true` line. 

I'm leaning toward not adding `@spec_files_loaded = true`.